### ISSUE TITLE
Implement Role Based Authorization

### ISSUE THREAD

**Issue by NicoDuvenage** [mojaloop/mojaloop#38]

**Comment by millerabel** [comment]
Refer to _[Notes on API Security Policy](https://www.dropbox.com/s/kgmc2rul58jhni9/Notes%20on%20API%20Security%20Policy.docx?dl=0)_ in the Non-Functional Requirements folder of the Level One Project Team Share Dropbox. I will bring this doc forward and convert to .md when we find a public home for design docs.

**Comment by mdebarros** [comment]
@millerabel: can the following document provide a guide line for the Possible Roles that can be implemented for this purpose: https://www.dropbox.com/home/Level%20One%20OSS%20Team%20Share/Discussion%20Docs/Operational%20Hub?preview=Operational+Hub+-+Feature+Priorities.xlsx

**Comment by kjw000** [comment]
Absolutely - this document along with the ppt in the same directory was created to socialize and provide context for the operational hub and the potential roles.    There is also the following document that helps to explain the roles in the XLS in more details: https://www.dropbox.com/s/ftala5ys1alumuw/Document%20Audience%20Roles.docx?dl=0
or - https://www.dropbox.com/preview/Level%20One%20OSS%20Team%20Share/discussion%20docs/Documentation/Document%20Audience%20Roles.docx?role=personal

**Comment by millerabel** [comment]
@mdebarros, yes that document and as @kjw000 outlines, are guides to thinking about roles. The document that I've provided (here: https://www.dropbox.com/s/kgmc2rul58jhni9/Notes%20on%20API%20Security%20Policy.docx?dl=0) look into a policy framework in which the roles might sit. We should be careful not to create roles that correspond to specific actions, without thinking about what multiple actions a person in a given role might perform. And yet, there will be cases where synthetic roles are appropriate (synthetic roles, where more than one authenticated individual must come together to enter into certain high-risk activities).
